File: cdm/java/mil/tatrc/physiology/datamodel/patient/actions/SEUrinate.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package mil.tatrc.physiology.datamodel.patient.actions;

import com.kitware.physiology.cdm.PatientActions.UrinateData;

public class SEUrinate extends SEPatientAction
{
  public SEUrinate()
  {

  }
  
  public void copy(SEUrinate other)
  {
    if(this==other)
      return;
    super.copy(other);    
  }
  
  public void reset()
  {
    super.reset();
  }
  
  public boolean isValid()
  {
    return true;
  }
  
  public static void load(UrinateData src, SEUrinate dst)
  {
    SEPatientAction.load(src.getPatientAction(), dst);
  }
  
  public static UrinateData unload(SEUrinate src) 
  {
    UrinateData.Builder dst = UrinateData.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  
  protected static void unload(SEUrinate src, UrinateData.Builder dst)
  {
    SEPatientAction.unload(src,dst.getPatientActionBuilder());
  }
  
  
  public String toString() 
  {
      return "Urinate";
  }
}

File: cdm/cpp/patient/actions/SEUrinate.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "patient/actions/SEUrinate.h"

SEUrinate::SEUrinate() : SEPatientAction()
{

}

SEUrinate::~SEUrinate()
{
  Clear();
}

void SEUrinate::Clear()
{
  SEPatientAction::Clear();
}

bool SEUrinate::IsValid() const
{
  return SEPatientAction::IsValid();
}

bool SEUrinate::IsActive() const
{
  return IsValid();
}

void SEUrinate::Load(const cdm::UrinateData& src, SEUrinate& dst)
{
  SEUrinate::Serialize(src, dst);
}
void SEUrinate::Serialize(const cdm::UrinateData& src, SEUrinate& dst)
{
  SEPatientAction::Serialize(src.patientaction(), dst);
}

cdm::UrinateData* SEUrinate::Unload(const SEUrinate& src)
{
  cdm::UrinateData* dst = new cdm::UrinateData();
  SEUrinate::Serialize(src, *dst);
  return dst;
}
void SEUrinate::Serialize(const SEUrinate& src, cdm::UrinateData& dst)
{
  SEPatientAction::Serialize(src, *dst.mutable_patientaction());
}

void SEUrinate::ToString(std::ostream &str) const
{
  str << "Patient Action : Urinate"; 
  if(HasComment())
    str<<"\n\tComment: "<<m_Comment;
  str << std::flush;
}
